### DISCUSSION TITLE
How to Flash Tasmota on a Lilygo T-RelayS3 (6 relays)

### DISCUSSION THREAD

**Discussion by SimonFili** [arendst/Tasmota#19549] [Show and tell]
I managed to flash Tasmota on this Lilygo device, here are the steps.

1) I connected a USB to TTL directly on the TX/RX/GND pin on the Lilygo relay board
 - I'm not using the T-U2T adapter

2) Power-up the device while pressing the boot button
- Better here to have a good power supply connected to the DC input

3) go to : https://tasmota.github.io/install/

4) In the drop-down, choose : Tasmota32-S3 (QIO Flash/OPI PSRAM)

5) Connect to the USB port of your USB to TTL 

6) Flash (erase if needed)
 - Here you should be able to connect to the Wi-Fi hotspot and configure the wifi and connect to it

7) Then you need to compile with using those instructions : https://github.com/arendst/Tasmota/discussions/18745 

8) Add those line to user_config_override.h 
#define USE_SHIFT595
#define USE_RTC_CHIPS               
#define USE_DS3231                 
#define USE_RTC_ADDR    0x51

9) Upload the new firmware with the Tasmota web interface

10) With the updated Firmware, you need to add a new template and apply it
Go copy it from https://templates.blakadder.com/lilygo_T-Relay-S3.html

11) You should now have a fully functional T-RelayS3 with Tasmota

**Comment by marc-gist** [comment]
it uses a PCF8563 for the RTC, so why include USE_DS3231?

**Reply by SimonFili** [reply]
This information is from the Template: https://templates.blakadder.com/lilygo_T-Relay-S3.html

I've not used the RTC chip on this board, so I cannot comment more on it.

**Comment by diegozalez** [comment]
Hi there is a problem with the new revision boards 
![image](https://github.com/user-attachments/assets/54fb40ce-a86c-4fdd-ac82-2a2fefc5f3fc)
As you can see enable the pin of the HT74HC595ARZ I pull high and it makes it really frustrating because if you follow all your steps it looks like everything is working but it is not.

I did a quick fix where I set GPIO 4 as a relay and if you turn it on it disables the relays and low to activate the output.

Do you suggest another way and maybe modify the template?

Thanks for your time it helps a lot.

**Reply by sfromis** [reply]
If what you want is to have gpio4 permanently low (without having a `Relay` config), you could config it as `Output Lo`. Or `Output Hi` for the opposite. Or do you mean having to change state of the pin?

**Reply by diegozalez** [reply]
Yes I have try that, it works but I guess it is not the best practice. I wonder if they made this change in the revision they must have a reason and I can't find a use for it.

**Reply by vish022091** [reply]
Do you mind sharing a screenshot of the Template you're using?

I tried the same but with no luck [I seem to have the new rev. board too]. Thanks.

**Reply by diegozalez** [reply]
![image](https://github.com/user-attachments/assets/67080b29-0559-4cdc-afcf-f73ca92a1194)
Yessss sure no problem, one other thing is that it works better in 2.4ghz wifi on channel 1
